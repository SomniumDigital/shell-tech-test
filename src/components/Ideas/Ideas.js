import React from 'react';
import EditableIdea from '../EditableIdea/EditableIdea';

const Ideas = ({
  ideasList,
  focusedIdea,
  edit,
  editHandler,
  editedIdea,
  updateInput,
  updateIdea
}) => (
  <ul>
    {ideasList.map((idea) => {
      const cssClass = focusedIdea === idea.id ? 'focusIdea idea' : 'idea';
      return (
        <li key={idea.id} className={cssClass}>
            <EditableIdea
              idea={idea}
              edit={edit}
              editIdea={editHandler}
              editedIdea={editedIdea}
              focusedIdea={focusedIdea}
              updateInput={updateInput}
              updateIdea={updateIdea}
            />
        </li>
      )}
    )}
  </ul>
);

export default Ideas
