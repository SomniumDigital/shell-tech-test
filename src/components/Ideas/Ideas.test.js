import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Ideas from './Ideas';

configure({ adapter: new Adapter() });

describe('Component: Ideas', () => {
  it('renders correctly', () => {
    const ideasListData = [{
      id: '123ABC',
      title: 'Test title',
      body: 'This is a test'
    }]
    const component = shallow(
      <Ideas
        ideasList={ideasListData}
      />
    )
    expect(component).toMatchSnapshot();
  });
});
