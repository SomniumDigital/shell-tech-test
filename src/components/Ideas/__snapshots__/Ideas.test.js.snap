// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Ideas renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Ideas
    ideasList={
        Array [
            Object {
              "body": "This is a test",
              "id": "123ABC",
              "title": "Test title",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="idea"
>
          <EditableIdea
                    edit={undefined}
                    editIdea={undefined}
                    editedIdea={undefined}
                    focusedIdea={undefined}
                    idea={
                              Object {
                                        "body": "This is a test",
                                        "id": "123ABC",
                                        "title": "Test title",
                                      }
                    }
                    updateIdea={undefined}
                    updateInput={undefined}
          />
</li>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "123ABC",
        "nodeType": "host",
        "props": Object {
          "children": <EditableIdea
            edit={undefined}
            editIdea={undefined}
            editedIdea={undefined}
            focusedIdea={undefined}
            idea={
                        Object {
                                    "body": "This is a test",
                                    "id": "123ABC",
                                    "title": "Test title",
                                  }
            }
            updateIdea={undefined}
            updateInput={undefined}
/>,
          "className": "idea",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "edit": undefined,
            "editIdea": undefined,
            "editedIdea": undefined,
            "focusedIdea": undefined,
            "idea": Object {
              "body": "This is a test",
              "id": "123ABC",
              "title": "Test title",
            },
            "updateIdea": undefined,
            "updateInput": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="idea"
>
            <EditableIdea
                        edit={undefined}
                        editIdea={undefined}
                        editedIdea={undefined}
                        focusedIdea={undefined}
                        idea={
                                    Object {
                                                "body": "This is a test",
                                                "id": "123ABC",
                                                "title": "Test title",
                                              }
                        }
                        updateIdea={undefined}
                        updateInput={undefined}
            />
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "123ABC",
          "nodeType": "host",
          "props": Object {
            "children": <EditableIdea
              edit={undefined}
              editIdea={undefined}
              editedIdea={undefined}
              focusedIdea={undefined}
              idea={
                            Object {
                                          "body": "This is a test",
                                          "id": "123ABC",
                                          "title": "Test title",
                                        }
              }
              updateIdea={undefined}
              updateInput={undefined}
/>,
            "className": "idea",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "edit": undefined,
              "editIdea": undefined,
              "editedIdea": undefined,
              "focusedIdea": undefined,
              "idea": Object {
                "body": "This is a test",
                "id": "123ABC",
                "title": "Test title",
              },
              "updateIdea": undefined,
              "updateInput": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
